drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -280,7 +280,10 @@ static int nrf5_set_channel(const struct device *dev, uint16_t channel)
 		return channel < 11 ? -ENOTSUP : -EINVAL;
 	}
 
-	nrf_802154_channel_set(channel);
+	if (nrf5_data.channel != channel) {
+		nrf_802154_channel_set(channel);
+		nrf5_data.channel = channel;
```

Comment:
If you initialize this variable to `nrf_802154_channel_get()` then it should be possible to safely use it instead of `nrf_802154_channel_get()` for example in:
https://github.com/zephyrproject-rtos/zephyr/blob/793c507209eb508c1c29441b56d60303df4187a7/drivers/ieee802154/ieee802154_nrf5.c#L511-L521

Reply:
do you think initializing it after `nrf_802154_init()` in `nrf5_init()` would be a good place?

Reply:
Yes, exactly. I would put it together with the following lines:
https://github.com/zephyrproject-rtos/zephyr/blob/793c507209eb508c1c29441b56d60303df4187a7/drivers/ieee802154/ieee802154_nrf5.c#L751-L754

Reply:
sounds good! Added that. I've decided to keep `nrf_802154_channel_get()` in debug logs- we don't need to optimize them, and IMO it would be easier to check if the channel on Radio's side is correct this way

Reply:
Sure, agreed.